lib/src/node-sass/render.ts

```diff
@@ -0,0 +1,215 @@
+  omitSourceMapUrl?: boolean;
+  outFile?: string | null;
+  sourceMap?: boolean | string;
+  // TODO(awjin): https://github.com/sass/embedded-protocol/issues/46
+  // sourceMapContents?: boolean;
+  sourceMapEmbed?: boolean;
+  sourceMapRoot?: string;
+  // TODO(awjin): https://github.com/sass/embedded-host-node/issues/13
+  // importer
+  // includePaths
+  // indentedSyntax
+  // functions
+  // outputStyle
+}
+
+/**
+ * The result returned by render().
+ *
+ * This attempts to match the Node Sass result object as closely as possible
+ * (see: https://github.com/sass/node-sass#result-object).
+ */
+export interface RenderResult {
+  css: Buffer;
+  map?: Buffer;
+  stats: {
+    start: number;
+    end: number;
+    duration: number;
+    entry: string;
+    // TODO(awjin): https://github.com/sass/embedded-protocol/issues/46
+    // includedFiles?: string[];
+  };
+}
+
+/**
+ * An error thrown by render().
+ *
+ * This attempts to match the Node Sass error object as closely as possible
+ * (see: https://github.com/sass/node-sass#error-object).
+ */
+export interface RenderError extends Error {
+  message: string;
+  status: number;
+  line?: number;
+  column?: number;
+  file?: string;
+}
+
+/**
+ * Converts Sass to CSS.
+ *
+ * This attempts to match the Node Sass `render()` API as closely as possible
+ * (see: https://github.com/sass/node-sass#usage).
+ */
+export async function render(
+  options: RenderOptions,
+  callback: (error?: RenderError, result?: RenderResult) => void
+): Promise<void> {
+  const file = options.file ? p.resolve(options.file) : undefined;
+  if (file === undefined && options.data === undefined) {
+    callback(
+      newRenderError(Error('Either options.data or options.file must be set.'))
+    );
+    return;
+  }
+
+  let sourceMap;
+  const getSourceMap = wasSourceMapRequested(options)
+    ? (map: RawSourceMap) => (sourceMap = map)
+    : undefined;
+
+  try {
+    const start = Date.now();
+    const css = options.data
+      ? await compileString({
+          source: options.data,
+          sourceMap: getSourceMap,
+          url: isEmpty(options.file) ? 'stdin' : pathToFileURL(options.file!),
+        })
+      : await compile({path: file!, sourceMap: getSourceMap});
+    callback(undefined, newRenderResult(options, start, css, sourceMap));
+  } catch (error) {
+    callback(newRenderError(error));
+  }
+}
+
+// Determines whether a sourceMap was requested by the call to render().
+function wasSourceMapRequested(options: RenderOptions): boolean {
+  return (
+    typeof options.sourceMap === 'string' ||
+    (options.sourceMap === true && !isEmpty(options.outFile))
+  );
+}
+
+// Transforms the compilation result into an object that mimics the Node Sass
+// API format.
+function newRenderResult(
+  options: RenderOptions,
+  start: number,
+  css: string,
+  sourceMap?: RawSourceMap
+): RenderResult {
+  const end = Date.now();
+  let sourceMapBytes;
+
+  if (wasSourceMapRequested(options)) {
```

Comment:
Can you just check `if (sourceMap)` here? It's only going to be set if `options` requests it, right?

Reply:
True, this double check is unnecessary.